context/app/static/js/components/cells/DatasetsSelectedByExpression/hooks.tsx

```diff
@@ -0,0 +1,170 @@
+  queryType: state.queryType,
+  setIsLoading: state.setIsLoading,
+});
+
+function getSearchQuery(cellsResults: { uuid: string }[]) {
+  return {
+    size: 10000,
+    post_filter: {
+      bool: {
+        must: [
+          {
+            term: {
+              'entity_type.keyword': 'Dataset',
+            },
+          },
+          {
+            terms: {
+              uuid: cellsResults.map((result) => result.uuid),
+            },
+          },
+        ],
+      },
+    },
+    _source: [
+      'uuid',
+      'hubmap_id',
+      'mapped_data_types',
+      'origin_samples_unique_mapped_organs',
+      'donor',
+      'last_modified_timestamp',
+    ],
+  };
+}
+
+function buildHitsMap(hits: { _id: string }[]) {
+  return hits.reduce(
+    (acc, hit) => {
+      acc[hit._id] = hit;
+      return acc;
+    },
+    {} as Record<string, { _id: string }>,
+  );
+}
+
+function useDatasetsSelectedByExpression() {
+  const { completeStep } = useAccordionStep();
+  const [message, setMessage] = useState<string | null>(null);
+  const { elasticsearchEndpoint, groupsToken } = useAppContext();
+  const [genomicModality, setGenomicModality] = useState('rna');
+  const resetFetchedUUIDs = useCellsChartLoadingStore(chartsStoreSelector);
+
+  const {
+    setResults,
+    minExpressionLog,
+    setMinExpressionLog,
+    minCellPercentage,
+    setMinCellPercentage,
+    cellVariableNames,
+    setCellVariableNames,
+    queryType,
+    setIsLoading,
+  } = useStore(cellsStoreSelector);
+
+  const { toastError } = useSnackbarActions();
+
+  function handleSelectModality(event: React.ChangeEvent<HTMLInputElement>) {
+    setGenomicModality(event.target.value);
+  }
+
+  const handleSubmit = useCallback(async () => {
+    setIsLoading(true);
+    setResults([]);
+    resetFetchedUUIDs();
+    const queryParams = {
+      type: queryType,
+      cellVariableNames,
+      minExpression: 10 ** minExpressionLog,
+      minCellPercentage,
+      modality: queryType === 'gene' ? genomicModality : undefined,
+    };
+    try {
+      completeStep(
+        <>
+          {cellVariableNames.join(', ')} | Expression Level 10<sup>{minExpressionLog}</sup> | {minCellPercentage}% Cell
+          Percentage
+        </>,
+      );
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const serviceResults = await new CellsService().getDatasets(queryParams);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-argument
+      const searchResults = await fetchSearchData(getSearchQuery(serviceResults), elasticsearchEndpoint, groupsToken);
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
+      const hitsMap = buildHitsMap(searchResults.hits.hits);
+      setResults(
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
```

Comment:
Are all of the `eslint-disable` due to the functions not being converted to typescript? Perhaps organize the `eslint-disable` into a block and leave a comment?

Reply:
Yup, these are all due to additional TS conversions; I've reorganized the comments into a single block disable/enable and added an explanatory TODO.